csharp/SSEtoRserve/RServeEvaluator.cs

```
@@ -456,49 +451,161 @@ private void HandleZeroRowsFromRserve(RserveConnection rserveConn)
+                {
+                    if (String.IsNullOrEmpty(resultDataColumns[col].Name))
+                    {
+                        tableDesc.Fields.Add(new FieldDescription
+                        {
+                            DataType = resultDataColumns[col].DataType
+                        });
+                    }
+                    else
+                    {
+                        tableDesc.Fields.Add(new FieldDescription
+                        { DataType = resultDataColumns[col].DataType,
+                          Name = resultDataColumns[col].Name
+                        });
+                    }
+                }
+                
+                var tableMetadata = new Metadata
+                {
+                    { new Metadata.Entry("qlik-tabledescription-bin", MessageExtensions.ToByteArray(tableDesc)) }
+                };
+                if (!cacheResultInQlik)
+                {
+                    tableMetadata.Add("qlik-cache", "no-store");
+                }
+                
+                await context.WriteResponseHeadersAsync(tableMetadata);
+
+                // Send data
+                var bundledRows = new BundledRows();
+
+                for (int i = 0; i < nrOfRows; i++)
                 {
                     var row = new Row();
-                    row.Duals.Add(new Dual() {  StrData = strings[i]??"" });
+                    for (int col = 0; col < nrOfCols; col++)
+                    {
+                        if (resultDataColumns[col].DataType == DataType.Numeric)
+                        {
+                            row.Duals.Add(new Dual() { NumData = resultDataColumns[col].Numerics[i] });
+                        }
+                        else if (resultDataColumns[col].DataType == DataType.String)
+                        {
+                            row.Duals.Add(new Dual() { StrData = resultDataColumns[col].Strings[i] ?? "" });
+                        }
```

Do you want to support returning dual data? (if set in tableDescription) If not, you might want to throw an error.

---

No, we only get one type from Rserve so it will not be possible. I will think about what should be done.

---

I have tested to set the return values in all the defined functions to Dual, then run the example when only getting numerics or strings from Rserve. We actually send an error from the plugin to Qlik in this case since the data type of the first column differs from the definition (see code lines 549-554). Therefore no changes needed, all works as expected.